scripts/generic-worker-windows/01-install-everything.ps1

```diff
@@ -0,0 +1,220 @@
+
+# download cygwin (not required, but useful)
+$client.DownloadFile("https://www.cygwin.com/setup-x86_64.exe", "C:\cygwin-setup-x86_64.exe")
+
+# install cygwin
+# complete package list: https://cygwin.com/packages/package_list.html
+Start-Process "C:\cygwin-setup-x86_64.exe" -ArgumentList "--quiet-mode --wait --root C:\cygwin --site http://cygwin.mirror.constant.com --packages openssh,vim,curl,tar,wget,zip,unzip,diffutils,bzr" -Wait -NoNewWindow -PassThru -RedirectStandardOutput "C:\cygwin_install.log" -RedirectStandardError "C:\cygwin_install.err"
+
+# open up firewall for ssh daemon
+New-NetFirewallRule -DisplayName "Allow SSH inbound" -Direction Inbound -LocalPort 22 -Protocol TCP -Action Allow
+
+# workaround for https://www.cygwin.com/ml/cygwin/2015-10/msg00036.html
+# see:
+#   1) https://www.cygwin.com/ml/cygwin/2015-10/msg00038.html
+#   2) https://cygwin.com/git/gitweb.cgi?p=cygwin-csih.git;a=blob;f=cygwin-service-installation-helper.sh;h=10ab4fb6d47803c9ffabdde51923fc2c3f0496bb;hb=7ca191bebb52ae414bb2a2e37ef22d94f2658dc7#l2884
+$env:LOGONSERVER = "\\" + $env:COMPUTERNAME
+
+# configure sshd (not required, but useful)
+Start-Process "C:\cygwin\bin\bash.exe" -ArgumentList "--login -c `"ssh-host-config -y -c 'ntsec mintty' -u 'cygwinsshd' -w 'qwe123QWE!@#'`"" -Wait -NoNewWindow -PassThru -RedirectStandardOutput "C:\cygrunsrv.log" -RedirectStandardError "C:\cygrunsrv.err"
+
+# start sshd
+Start-Process "net" -ArgumentList "start sshd" -Wait -NoNewWindow -PassThru -RedirectStandardOutput "C:\net_start_sshd.log" -RedirectStandardError "C:\net_start_sshd.err"
+
+# download bash setup script
+$client.DownloadFile("https://raw.githubusercontent.com/petemoore/myscrapbook/master/setup.sh", "C:\cygwin\home\Administrator\setup.sh")
+
+# run bash setup script
+Start-Process "C:\cygwin\bin\bash.exe" -ArgumentList "--login -c 'chmod a+x setup.sh; ./setup.sh'" -Wait -NoNewWindow -PassThru -RedirectStandardOutput "C:\Administrator_cygwin_setup.log" -RedirectStandardError "C:\Administrator_cygwin_setup.err"
+
+# install dependencywalker (useful utility for troubleshooting, not required)
+md "C:\DependencyWalker"
+Expand-ZIPFile -File "C:\depends22_x64.zip" -Destination "C:\DependencyWalker" -Url "http://dependencywalker.com/depends22_x64.zip"
+
+# install ProcessExplorer (useful utility for troubleshooting, not required)
+md "C:\ProcessExplorer"
+Expand-ZIPFile -File "C:\ProcessExplorer.zip" -Destination "C:\ProcessExplorer" -Url "https://download.sysinternals.com/files/ProcessExplorer.zip"
+
+# install ProcessMonitor (useful utility for troubleshooting, not required)
+md "C:\ProcessMonitor"
+Expand-ZIPFile -File "C:\ProcessMonitor.zip" -Destination "C:\ProcessMonitor" -Url "https://download.sysinternals.com/files/ProcessMonitor.zip"
```

Comment:
A lot of these steps aren't required for general purpose worker types, and were specific to the needs of some of the gecko worker types, for example. Can we separate the required steps for installing generic-worker from the other steps which may be specific to individual worker types? Since we don't have containers on Windows I suspect the setup steps will always be quite varied between individual worker types, depending on the needs of the tasks that run on them.